server/http.go

```diff
@@ -14,6 +15,11 @@ import (
 	"github.com/gorilla/mux"
 )
 
+// Certificate is a PEM-encoded cert/key pair
```

Comment:
Move to types.go and Use for eebus, too?

Reply:
yes and no. I used the type `string` whereas eebus uses `[]byte`. I tried []byte for my case but had type conversion problems from yaml. Any idea how to convert a string into []byte upon decoding?

Reply:
You are both passing the same content- this should work.

Reply:
we use different code. EEbus decoded into `map[string]interface{}` and later into a struct, while I'm decoding directly into a struct. I'm not sure why it works for eebus but I'm failing in `viper.Unmarshal()`. I'll try to play with unmarshaling options. If I won't successes I'll see if it's possible to switch eebus's code into using a string instead of []byte.

Reply:
okay, I've looked into how eebus decodes things. It uses a customized decoding config in particular `mapstructure.TextUnmarshallerHookFunc()` which is part of the `util.DecodeOther()` utility. I can apply `TextUnmarshallerHookFunc` also in the available general config decoding (`loadConfigFile()`) however I'm not sure about the consequences. I think changing the type to `string` of the certificate's public/private is a less invasive approach. However, happy to go either way based on your opinion.

Potential diff for the decoding in `loadConfigFile()`:
```diff
        if err == nil {
-               if err = viper.UnmarshalExact(&conf); err != nil {
+               opts := func(conf *mapstructure.DecoderConfig) {
+                       conf.DecodeHook = mapstructure.ComposeDecodeHookFunc(
+                               mapstructure.StringToTimeDurationHookFunc(),
+                               mapstructure.TextUnmarshallerHookFunc(),
+                       )
+               }
+
+               if err = viper.UnmarshalExact(&conf, opts); err != nil {
                        err = fmt.Errorf("failed parsing config file: %w", err)
                }
        }
```

Reply:
@andig can you please have a look? I need your opinion before moving forward.

Reply:
>I'm not sure why it works for eebus but I'm failing in viper.Unmarshal()

What is failing? The TextUnmarshaller shouldn't do anything.


> If I won't successed I'll see if it's possible to switch eebus's code into using a string instead of []byte.

Either should be convertible my simply type-casting.

Reply:
> What is failing? The TextUnmarshaller shouldn't do anything.

it tries to cast the string into []int8:
```
cannot parse 'Network.Certificate.Private[0]' as uint: strconv.ParseUint: parsing "-----BEGIN EC PRIVATE KEY-----
```

> > If I won't successed I'll see if it's possible to switch eebus's code into using a string instead of []byte.

> Either should be convertible my simply type-casting.

If string -> []byte casting is fine, we can merge existing code. It does exactly that.

Reply:
@andig ?

Reply:
I dont know why it does that and why it doesnt for eebus, sorry.